### DISCUSSION TITLE
Does not run on ARMv6 (Raspberry Pi 1)

### DISCUSSION THREAD

**Discussion by uniquepurpose** [AsamK/signal-cli#443] [Q&A] [answered]
Hi,

I'm running a Raspberry Pi on raspios from 2021-01-11 (Buster, 10.7). I have tried versions 8, 9, 10 and 11 of openjdk but cannot get signal-cli 0.7.3 to run.
I get this message:

> Error occurred during initialization of VM
> Server VM is only supported on ARMv7+ VFP

Except when selecting "/usr/lib/jvm/java-9-openjdk-armhf/bin/java" with "sudo update-alternatives --config java". Then I get:

> Error: A JNI error has occurred, please check your installation and try again
> Exception in thread "main" java.lang.UnsupportedClassVersionError: org/asamk/signal/Main has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
>         at java.lang.ClassLoader.defineClass1(Native Method)
>         at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
>         at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
>         at java.net.URLClassLoader.defineClass(URLClassLoader.java:468)
>         at java.net.URLClassLoader.access$100(URLClassLoader.java:74)
>         at java.net.URLClassLoader$1.run(URLClassLoader.java:369)
>         at java.net.URLClassLoader$1.run(URLClassLoader.java:363)
>         at java.security.AccessController.doPrivileged(Native Method)
>         at java.net.URLClassLoader.findClass(URLClassLoader.java:362)
>         at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
>         at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
>         at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
>         at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:495)

Is there any way to run signal-cli on ARMv6 processors?

**Comment by AsamK** [comment, marked as answer]
signal-cli requires at least java 11 so versions definitely won't work.
It seems your openjdk build doesn't support armv6  (Server VM is only supported on ARMv7+ VFP), you'll need to find a build that does.
Maybe this could help: https://medium.com/javarevisited/how-to-install-and-use-java-11-and-javafx-11-on-raspberry-pi-boards-with-armv6-processor-ca7109b76b7c
